extensions/2.0/Khronos/KHR_interactivity/Specification.adoc

```diff
@@ -0,0 +1,2914 @@
+
+The `values` array defines IDs and type indices of the sockets associated with the event. If the array is undefined, the event has no associated value sockets.
+
+== Variables
+
+The `variables` array defines custom variables with their types and optional initialization values.
+
+The following example defines a custom variable with its initial value:
+
+```json
+"variables": [
+  {
+    "type": 0,
+    "value": [0.5,  0.5]
+  }
+]
+```
+
+The `type` value defines the index of the variable type.
+
+The `value` array, if present, defines the initial variable value. The following table defines array size and default values for all value types defined in this extension.
+
+[cols="1,1,2", options="header"]
+|===
+|  Type      | Array size | Default value
+| `bool`     |          1 | Boolean false
+| `float`    |          1 | Floating-point zero
+| `float2`   |          2 | Two floating-point zeros
+| `float3`   |          3 | Three floating-point zeros
+| `float4`   |          4 | Four floating-point zeros
+| `float4x4` |         16 | Sixteen floating-point zeros
+| `int`      |          1 | Integer zero
+|===
+
+If the variable type is custom, the `value` property is defined by the extension defining the custom type.
+
+== Nodes
+
+The `nodes` array defines the behavior graph.
+
+Each element of the `nodes` array represents a node instance, i.e., it specifies node's type, configuration, sources of input value sockets, and pointers of the output flow sockets.
+
+Input value sockets **MAY** have inline constant values; in this case, the value socket type **MUST** be defined.
+
+Inline values and configurations use JSON arrays similarly to the initial variable values.
+
+The following example instantiates a `math/add` node that has both its input value sockets filled with inline integer values.
+
+```json
+"nodes": [
+  {
+    "type": "math/add",
+    "values": [
+      {
+        "id": "a",
+        "value": [1],
+        "type": 1
+      },
+      {
+        "id": "b",
+        "value": [2],
+        "type": 1
+      }
+    ]
+  }
+]
+```
+
+The following example instantiates three nodes. The `math/sub` node has both its input value sockets connected to output value sockets of two other nodes: `math/pi` and `math/e`.
+
+```json
+"nodes": [
+  {
+    "type": "math/pi"
+  },
+  {
+    "type": "math/e"
+  },
+  {
+    "type": "math/sub",
+    "values": [
+      {
+        "id": "a",
+        "node": 0,
+        "socket": "value"
+      },
+      {
+        "id": "b",
+        "node": 1,
+        "socket": "value"
+      }
+    ]
+  }
+]
+```
+
+The following example instantiates two nodes. The `variable/set` node sets a custom variable with index `0` when the start event happens.
+
+```json
+"variables": [
+  {
+    "type": 1
+  }
+],
+"nodes": [
+  {
+    "type": "variable/set",
+    "configuration": [
+      {
+        "id": "variable",
+        "value": [0]
+      }
```

Comment:
are configuration values implicitly typed in JSON schema or they should have "type" key as well?